### ISSUE TITLE
Select :options behavior change not documented

### ISSUE THREAD

**Issue by PatrickKennedy** [bootstrap-vue/bootstrap-vue#1098]
In the recent update the documented method of passing an object to b-form-select has changed.

https://bootstrap-vue.js.org/docs/components/form-select 
in the Options property under Object it states:
"Keys are mapped to value and values are mapped to option object."

However, that behavior was changed to return undefined if the valueField doesn't exist: 
https://github.com/bootstrap-vue/bootstrap-vue/commit/0a50398cd03e9b016ea1c87a82a79268f8e5945a#diff-af4f3d5f783a2851de8bd042b34dd5aeL32

To me it seems like an odd change as it means the object key is expected to be human readable as opposed to being a more friendly backend value that make it easy to tie a data object together with the value from a select. With the new model to retain that coupling you'd need to duplicate the key value.

It's a small change in the data, but still decreases the benefit of using objects.

EDIT: A working JSFiddle with an example of coupling with an object:
https://jsfiddle.net/16ecp78w/2/
Note: The additional  `value` element in the data

**Comment by tmorehouse** [comment]
Hmmm...

I think this was because in the original alpha branch, if a user wanted a value of `0` or `false`, it would get overwritten with they `key` instead of the actual value they wanted (i.e. `0` or `false`, which both loosely translate to `false`):

```js
// original alpha code
option.value = option[this.valueField] || key;
```

We may have to revisit this again and throw in some additional tests to check for `falsey` values.

**Comment by tmorehouse** [comment]
This should be fixed in the upcoming release v1.0.0.beta.10
